pkg/corerp/frontend/controller/secretstores/kubernetes.go

```diff
@@ -179,11 +180,14 @@ func fromResourceID(id string) (ns string, name string, err error) {
 // UpsertSecret creates or updates a Kubernetes secret based on the incoming request and returns the secret's location in
 // the output resource.
 func UpsertSecret(ctx context.Context, newResource, old *datamodel.SecretStore, options *controller.Options) (rest.Response, error) {
+	logger := ucplog.FromContextOrDiscard(ctx)
 	ref := newResource.Properties.Resource
 	if ref == "" && old != nil {
 		ref = old.Properties.Resource
 	}
-
+	if isGlobalScopedResource(newResource) && newResource.Properties.Resource == "" {
```

Comment:
Do we expect resource to be non-empty even if this is an update? If not, then this check should use `ref`

```suggestion
	if isGlobalScopedResource(newResource) && ref == "" {
```

Reply:
you are right! need to check use `ref`. Thanks for pointing this